app/src/main/java/io/homeassistant/companion/android/sensors/SensorsSettingsFragment.kt

```diff
@@ -59,29 +60,40 @@ class SensorsSettingsFragment : PreferenceFragmentCompat() {
         SensorReceiver.MANAGERS.sortedBy { it.name }.forEach { manager ->
             val prefCategory = PreferenceCategory(preferenceScreen.context)
             prefCategory.title = getString(manager.name)
-            preferenceScreen.addPreference(prefCategory)
-            manager.availableSensors.sortedBy { it.name }.forEach { basicSensor ->
+            var hasSensor: Boolean
+            val packageManager: PackageManager? = context?.packageManager
+            hasSensor = when (manager.name) {
+                R.string.sensor_name_light -> packageManager!!.hasSystemFeature(PackageManager.FEATURE_SENSOR_LIGHT)
+                R.string.sensor_name_pressure -> packageManager!!.hasSystemFeature(PackageManager.FEATURE_SENSOR_BAROMETER)
+                R.string.sensor_name_proximity -> packageManager!!.hasSystemFeature(PackageManager.FEATURE_SENSOR_PROXIMITY)
+                R.string.sensor_name_steps -> packageManager!!.hasSystemFeature(PackageManager.FEATURE_SENSOR_STEP_COUNTER)
+                else -> true
+            }
```

Comment:
We should add a function to SensorManager (`hasSensor(context)`)that checks for this and defaults to true and override in the needed managers.

Reply:
ah so thats how we bring context in 😃 thank you!